chiselFrontend/src/main/scala/chisel3/core/BlackBox.scala

```diff
@@ -52,22 +130,29 @@ case class RawParam(value: String) extends Param
   * }}}
   * @note The parameters API is experimental and may change
   */
-abstract class BlackBox(val params: Map[String, Param] = Map.empty[String, Param]) extends Module {
+abstract class BlackBox(val params: Map[String, Param] = Map.empty[String, Param]) extends BaseBlackBox {
+  def io: Record
```

Comment:
You are refactoring the module API, now is the time to address improper existing behavior. (Regardless, what prior modules are able to reliably get away with a non-stable io?)